src/utils/calculations.ts

```diff
@@ -14,7 +13,22 @@ interface CalculateSharesAmountProps {
 export function calculateSharesAmount(props: CalculateSharesAmountProps): Wei {
   const { amount, decimals, pricePerShare } = props;
   if (amount.isNaN()) return '0';
-  const sharePrice = formatUnits(pricePerShare, decimals);
+  const sharePrice = toUnit(pricePerShare, parseInt(decimals));
   const ONE_UNIT = toBN('10').pow(decimals);
-  return amount.dividedBy(sharePrice).multipliedBy(ONE_UNIT).toFixed(0);
+  return amount.div(sharePrice).times(ONE_UNIT).toFixed(0);
+}
+
+interface CalculateUnderlyingAmountProps {
+  decimals: string;
```

Comment:
would go a bit more descriptive with the props names like: yvDecimals or underlyingDecimals and so on to avoid confusion on what to send on each param.

Reply:
done